Bmx.CommandLine/IniConfiguration.cs

```diff
@@ -0,0 +1,80 @@
+﻿#nullable enable
+using System;
+using System.IO;
+using System.Security;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.FileProviders.Physical;
+
+namespace Bmx.CommandLine {
+	public class IniConfiguration : IBmxConfig {
+		public string? Org { get; }
+		public string? User { get; }
+		public string? Account { get; }
+		public string? Role { get; }
+		public string? Profile { get; }
+
+		public IniConfiguration() {
+			var config = GetConfiguration();
+
+			Org = config["org"];
+			User = config["user"];
+			Account = config["account"];
+			Role = config["role"];
+			Profile = config["profile"];
+		}
+
+		private static IConfiguration GetConfiguration() {
+			// Main config is at ~/.bmx/config
+			var configLocation = Path.Join( Environment.GetFolderPath( Environment.SpecialFolder.UserProfile ), ".bmx",
+				"config" );
+
+			// If set, we recursively look up from CWD (all the way to root) for additional bmx config files (labelled as .bmx)
+			var config = new ConfigurationBuilder().AddIniFile( configLocation, true );
```

Comment:
nit: name the parameter when passing in a literal for better readability. At least that's what's done in the LMS.
```suggestion
			var config = new ConfigurationBuilder().AddIniFile( configLocation, optional: true );
```

Reply:
similarly on L47.

Reply:
👍 that's a pretty cool feature.